### PR TITLE
Infrastructure fix: is_single_node attribute elimination from MPIR_Comm structure

### PR DESCRIPTION
**Pull request by AlexeyMalkhanov**
The PR #2935 has implementations of the additional compositions for the case when communicator resides inside one node. In this case, we can invoke one of SHM collectives on the whole communicator. That allowed to get rid of "is_single_node" attribute from MPIR_Comm structure that had been introduced to handle such cases.

### CODE REVIEW

**Comment by raffenet** [comment]
Always love a patch that deletes code.
test:jenkins/ch3/tcp
test:jenkins/ch4/most

**Comment by raffenet** [comment]
Sure, no hurry.

**Comment by jain-surabhi-23** [comment]
I am not using `is_single_node` anymore. Instead, relying on `comm->hierarchy_kind == MPIR_COMM_HIERARCHY_KIND__NODE`.

File: src/mpid/ch4/src/ch4_coll_impl.h [hunk 1]
```diff
@@ -82,21 +82,9 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_Barrier_composition_beta(MPIR_Comm * comm, MP
...
     void * barrier_container = MPIDI_coll_get_next_container(ch4_algo_parameters_container);
 
-#ifndef MPIDI_BUILD_CH4_SHM
     mpi_errno =
         MPIDI_NM_mpi_barrier(comm, errflag, barrier_container);
     if (mpi_errno) MPIR_ERR_POP(mpi_errno);
-#else
-    if(comm->is_single_node) {
-        mpi_errno =
-            MPIDI_SHM_mpi_barrier(comm, errflag, barrier_container);
```
[the remark this answers is not in the document]
Comment (AlexeyMalkhanov) [reply src/mpid/ch4/src/ch4_coll_impl.h]: @raffenet for SHM collectives we introduce a separate composition (these changes are implemented in #2935). I'll split the changes into two commits: one for ch4 and the other for a communicator attributes.

**Pull request merged**